daemon/cluster/executor/container/controller.go

```diff
@@ -52,7 +55,11 @@ func newController(b executorpkg.Backend, task *api.Task, secrets exec.SecretGet
 	}, nil
 }
 
+// Task returns the controller's task.
 func (r *controller) Task() (*api.Task, error) {
+	// concurrency safe
+	r.taskMu.RLock()
+	defer r.taskMu.RUnlock()
```

Comment:
seems fine to me either way unless there is some overhead associated with defer that i am unaware of.